zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -145,11 +146,13 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             0.expr(),
             None,
         );
+        let is_caller_callee_equal =
```

Comment:
Don't know if this is the correct way to do it.
cc/ @han0110

Reply:
Ah right, never thought about tx to self 🤦‍♂️

I think this would work, but to make it even easier we could have the line 155 to take a witness freely chosen by prover, because we don't really use its previous value here (we only need to make sure it's updated to `1`), but we still need to assign a valid value, otherwise the State circuit consistency check will fail (perhaps one of cause of https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1233)

Reply:
I can remove the IsEqualGadget if it doesn't impact soundness of the circuit 👌

Reply:
Is there a easy way to do a equality check on the Cell/Expression types? 😅

Reply:
you mean to check if 2 cells are identical as `Expression` structure? but why are we doing that here? i think we could do a `query_bool` for this witness.

Reply:
I looked at the constraint builder but I don't grok how `query_bool` would return callee==caller. I guess I have to lookup these values beforehand? Anyhow, given that the prover is free to choose those values I opted into select::expr given that 'looks' simpler. Please correct me if I'm going the wrong way :)

Reply:
nvm. After being refreshed, I now see my misunderstanding 😆